plugins/arangodbendorsements/main.go

```diff
@@ -335,54 +199,25 @@ func (e *ArangoEndorsementStore) GetSwIDTagsLinkedToHWTagFromDB(HwTag string) ([
 }
 
 // GetHardwareID is the query function to fetch the Hardware Id associated to a platform
-func (e *ArangoEndorsementStore) GetHardwareID(args common.QueryArgs) (common.QueryResult, error) {
+func (e *EndorsementStore) GetHardwareID(args common.QueryArgs) (common.QueryResult, error) {
 	var platformID string
 	var result []interface{}
-	var HardwareInstance HardwareIdentityWrapper
-	arangoConn := new(ArangoDBConnParams)
-
+	var HwID HardwareIdentity
+	hwMap := make(map[string]string)
+	//var HardwareWrapperArray []HardwareIdentityWrapper
 	platformID, err := ExtractPlatformID(args)
 	if err != nil {
 		// Handle Platform Id Error
 		return nil, fmt.Errorf("failed to extract platform_id from query params %v", err)
 	}
 	log.Printf("received platform Id=%s", platformID)
-
-	bindVars := map[string]interface{}{
-		"platformId": platformID,
-	}
-
-	ctx := context.Background()
-
-	if err = e.ConnectToArangoDB(ctx, arangoConn); err != nil {
-		return nil, err
-	}
-
-	query := "FOR d IN " + e.HwCollection
-	query += " FILTER d.HardwareIdentity.`psa-hardware-rot`.`implementation-id` == @platformId RETURN d"
-	log.Printf("input query = %s", query)
-
-	cursor, err := arangoConn.db.Query(ctx, query, bindVars)
+	// Fetch the HW ID container from the DB associated to this platform
+	err = e.GetHwIDFromDB(platformID, &HwID)
 	if err != nil {
-		return nil, fmt.Errorf("query failed with reason %v", err)
+		return nil, fmt.Errorf("failed to fetch from store %v", err)
```

Comment:
could be made more useful by adding context to the error message:

```suggestion
		return nil, fmt.Errorf("failed to fetch HW ID from store: %w", err)
```